### DISCUSSION TITLE
In the benchmarks why there is a difference between the hnswlib to milvus hnsw

### DISCUSSION THREAD

**Discussion by sandraplato1234** [milvus-io/milvus#4741] [Q&A and General discussion]
Hi 
I read the benchmarks details and I want to understand why there is such difference between the performance curves of hnswlib to milvus hnsw
specially in fashion-mnist-784-euclidean (k = 10)
Thanks in advance

**Comment by stephenleo** [comment]
[ANN-Benchmarks](https://github.com/erikbern/ann-benchmarks) is using an old version of Milvus (0.6): https://github.com/erikbern/ann-benchmarks/blob/3b99110d2616c67fe53f7b000cfe484f800b7174/install/Dockerfile.milvus#L2

I think it will be interesting to see the benchmarks on the 0.10.5 version of Milvus.

**Reply by sandraplato1234** [reply]
Ok thanks for the quick reply.
Another question is why milvus benchmarks does not include benchmarks on gpu should be better if fune tune parameters,am I right ?
Thanks

**Reply by stephenleo** [reply]
Yeah from the [ANN-Benchmarks](https://github.com/erikbern/ann-benchmarks) documentation it seems it is focussing mainly on CPU-based implementation to have a fair comparison.

"
We mainly support CPU-based ANN algorithms. GPU support exists for FAISS, but it has to be compiled with GPU support locally and experiments must be run using the flags --local --batch.
"
